apps/wide-deep-recommendation/dummy_data_generating.py

```diff
@@ -0,0 +1,144 @@
+                                for i in range(random.randint(0, 10))])
+    tweet_type = random.choices(tweet_list)[0]
+    language = random.choices(language_list)[0]
+    tweet_timestamp = random.randint(946656000, 1609430400)
+    engaged_with_user_id = "".join(random.choices(id_list, k=32))
+    engaged_with_user_follower_count = random.randint(0, 10000)
+    engaged_with_user_following_count = random.randint(0, 10000)
+    engaged_with_user_is_verified = bool(random.getrandbits(1))
+    engaged_with_user_account_creation = random.randint(946656000, 1609430400)
+    enaging_user_id = "".join(random.choices(id_list, k=32))
+    enaging_user_follower_count = random.randint(0, 10000)
+    enaging_user_following_count = random.randint(0, 10000)
+    enaging_user_is_verified = bool(random.getrandbits(1))
+    enaging_user_account_creation = random.randint(946656000, 1609430400)
+    engagee_follows_engager = bool(random.getrandbits(1))
+    reply = bool(random.getrandbits(1))
+    reply_timestamp = random.randint(946656000, 1609430400) if reply else None
+    retweet = bool(random.getrandbits(1))
+    retweet_timestamp = random.randint(
+        946656000, 1609430400) if retweet else None
+    comment = bool(random.getrandbits(1))
+    retweet_with_comment_timestamp = random.randint(
+        946656000, 1609430400) if comment else None
+    like = bool(random.getrandbits(1))
+    like_timestamp = random.randint(946656000, 1609430400) if like else None
+    return (test_tokens, hashtags, tweet_id, present_media, present_links, present_domains,
+            tweet_type, language, tweet_timestamp, engaged_with_user_id,
+            engaged_with_user_follower_count, engaged_with_user_following_count,
+            engaged_with_user_is_verified, engaged_with_user_account_creation,
+            enaging_user_id, enaging_user_follower_count, enaging_user_following_count,
+            enaging_user_is_verified, enaging_user_account_creation,
+            engagee_follows_engager, reply_timestamp, retweet_timestamp,
+            retweet_with_comment_timestamp, like_timestamp)
+
+
+if __name__ == '__main__':
+    args = _parse_args()
+
+    OrcaContext.log_output = True
+
+    executor_cores = 8
+    num_executor = 4
+    executor_memory = args.memory
+    driver_cores = 4
+    driver_memory = "12g"
+    conf = {"spark.network.timeout": "10000000",
+            "spark.sql.broadcastTimeout": "7200",
+            "spark.sql.shuffle.partitions": "2000",
+            "spark.locality.wait": "0s",
+            "spark.sql.crossJoin.enabled": "true",
+            "spark.task.cpus": "1",
+            "spark.executor.heartbeatInterval": "200s",
+            "spark.driver.maxResultSize": "40G",
+            "spark.eventLog.enabled": "true",
+            "spark.app.name": "recsys-dummy-data-generation",
+            "spark.debug.maxToStringFields": "100"}
+    sc = init_orca_context("yarn", cores=executor_cores,
```

Comment:
we also need to provide a cluster mode to be local, refer to https://github.com/intel-analytics/BigDL/blob/main/python/friesian/example/dien/dien_preprocessing.py#L68